MapleServer2/Types/Trophy.cs

```diff
@@ -0,0 +1,122 @@
+                return;
+            }
+            Counter += amount;
+
+            if (Counter >= Condition)
+            {
+                ProvideReward(session);
+                NextGrade++;
+                // level up but not completed
+                if (NextGrade <= MaxGrade)
+                {
+                    Condition = TrophyMetadataStorage.GetGrade(Id, NextGrade).Condition;
+                }
+                // level up and completed
+                else
+                {
+                    IsDone = true;
+                    NextGrade--;
+                    string[] cats = TrophyMetadataStorage.GetMetadata(Id).Categories;
+                    foreach (string cat in cats)
+                    {
+                        switch (cat)
+                        {
+                            case string s when s.Contains("combat"):
+                                session.Player.TrophyCount[0] += 1;
+                                break;
+                            case string s when s.Contains("adventure"):
+                                session.Player.TrophyCount[1] += 1;
+                                break;
+                            case string s when s.Contains("lifestyle"):
+                                session.Player.TrophyCount[2] += 1;
+                                break;
+                        }
+                    }
+                }
+                Timestamps.Add(DateTimeOffset.UtcNow.ToUnixTimeSeconds());
+            }
+        }
+
+        private void ProvideReward(GameSession session)
+        {
+            TrophyGradeMetadata grade = TrophyMetadataStorage.GetGrade(Id, NextGrade);
+            RewardType type = (RewardType) grade.RewardType;
+            switch (type)
+            {
+                case RewardType.Unknown:
+                    break;
+                case RewardType.itemcoloring:
+                    break;
+                case RewardType.shop_ride:
+                    break;
+                case RewardType.title:
+                    break;
+                case RewardType.beauty_hair:
+                    break;
+                case RewardType.statPoint:
+                    session.Player.StatPointDistribution.AddTotalStatPoints(grade.RewardValue, OtherStatsIndex.Trophy);
+                    session.Send(StatPointPacket.WriteTotalStatPoints(session.Player));
+                    break;
+                case RewardType.skillPoint:
+                    break;
+                case RewardType.beauty_makeup:
+                    break;
+                case RewardType.shop_build:
+                    break;
+                case RewardType.item:
+                    break;
+                case RewardType.shop_weapon:
+                    break;
+                case RewardType.dynamicaction:
+                    break;
+                case RewardType.etc:
+                    break;
+                case RewardType.beauty_skin:
+                    break;
```

Comment:
```suggestion
                case RewardType.Unknown:
                case RewardType.itemcoloring:
                case RewardType.shop_ride:
                case RewardType.title:
                case RewardType.beauty_hair:
                case RewardType.skillPoint:
                case RewardType.beauty_makeup:
                case RewardType.shop_build:
                case RewardType.item:
                case RewardType.shop_weapon:
                case RewardType.dynamicaction:
                case RewardType.etc:
                case RewardType.beauty_skin:
                    break;
                case RewardType.statPoint:
                    session.Player.StatPointDistribution.AddTotalStatPoints(grade.RewardValue, OtherStatsIndex.Trophy);
                    session.Send(StatPointPacket.WriteTotalStatPoints(session.Player));
                    break;
                default:
                    break;
```

Reply:
it will ultimately require breaks at every line so are you suggesting we should keep the code clean while I wait for other parts to be implemented?

Reply:
I think it's better to keep the code clean since we don't know when everything else will be implemented!